### DISCUSSION TITLE
umi4按照文档使用 pnpm 新建项目，启用 Prettier失败

### DISCUSSION THREAD

**Discussion by Sallyfafafa** [umijs/umi#10979] [Q&A]
使用 官方推荐 pnpm安装项目，在 `pnpm umi g`先是报找不到 umi 命令，我通过 `pnpm install umi -g`下载 umi @4.0.65 版本后，再次执行  `pnpm umi g`，如下图所示，没有出现[快速上手文档](https://umijs.org/docs/tutorials/getting-started#%E5%90%AF%E7%94%A8-prettier%E5%8F%AF%E9%80%89)中的 `✔ Pick generator type › Enable Prettier -- Enable Prettier`

<img width="955" alt="image" src="https://user-images.githubusercontent.com/40735740/232358168-6c0e4eae-d27f-4a40-8171-8bd21867b261.png">

**Comment by fz6m** [comment]
没能复现你说的，给一个不能用 `pnpm umi g` 的最小复现看看。

**Reply by Sallyfafafa** [reply]
我本地环境如下：

<img width="742" alt="image" src="https://user-images.githubusercontent.com/40735740/232647949-7902cfbb-881f-401d-b767-a8af7d12eb4d.png">

然后
```
mkdir myapp && cd myapp
pnpm dlx create-umi@latest
```
选择 Ant Design Pro 做模版：
<img width="857" alt="image" src="https://user-images.githubusercontent.com/40735740/232648868-6caa84af-982a-4d27-8ece-185c8d5f9b95.png">

这次 `pnpm umi g` 可以用，但是依然没有` ✔ Pick generator type › Enable Prettier -- Enable Prettier`

<img width="790" alt="image" src="https://user-images.githubusercontent.com/40735740/232649021-a58718f4-e6e9-4474-9c5d-0226ba895792.png">
<img width="798" alt="image" src="https://user-images.githubusercontent.com/40735740/232649319-613453c1-2f15-4671-9288-959c9708d4e0.png">

**Reply by fz6m** [reply]
选的是 max 项目模板，已经自带 prettier 配置文件了，就不会出现在这个列表了。
